### PR TITLE
add id set validations - script using real cmds, incident type has real default PB

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
- [x] Ready

## Description
Aside from the added validations there was code in the id.py that was not called `is_id_duplicated`,`is_file_has_used_id` - so I deleted him.

## Related Issues
fixes: https://github.com/demisto/etc/issues/29559

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by yaakovi — COMMENTED** [review]
looks ok,
@roysagi make sure to include this validation when migrating to content objects.

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -115,6 +115,7 @@
     "playbook_cant_have_deletecontext_all": "PB105",
     "using_instance_in_playbook": "PB106",
     "invalid_script_id": "PB107",
+    "invalid_script_using_commands": "SC100",
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:118]: we already have SC100 and 101:
image

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 2]
```diff
@@ -69,7 +67,7 @@ def load_id_set(self):
 
             return id_set
 
-    def is_valid_in_id_set(self, file_path: str, obj_data: OrderedDict, obj_set: list):
+    def _is_valid_in_id_set(self, file_path: str, obj_data: OrderedDict, obj_set: list):
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/id.py:70]: why is this change required?

File: demisto_sdk/commands/common/hook_validations/id.py [hunk 3]
```diff
@@ -118,128 +116,108 @@ def is_file_valid_in_set(self, file_path):
...
+        Args:
+            script_data (dict): Dictionary that holds the extracted details from the given script.
 
-        return is_used
+        Returns:
+            bool. Whether the script has a non real command.
+        """
+        depends_on_commands = script_data.get('depends_on')
+        for command in depends_on_commands:
+            if command != 'test-module' and not command.endswith('-test'):
```
Comment (roysagi) [inline demisto_sdk/commands/common/hook_validations/id.py:217]: why cant the command end with test?

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -779,6 +781,16 @@ def invalid_script_id(script_entry_to_check, pb_task):
                f"2 - The id_set.json file is up to date. Delete the file by running: rm -rf Tests/id_set.json and" \
                f" rerun the command."
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_script_using_commands(script_name, command):
```
Comment (roysagi) [inline demisto_sdk/commands/common/errors.py:786]: maybe a better name would be: `invalid_command_name_in_script`

File: demisto_sdk/commands/validate/validate_manager.py [hunk 5]
```diff
@@ -336,10 +336,17 @@ def run_validations_on_file(self, file_path, pack_error_ignore_list, is_modified
...
+            # if True:
             click.echo(f"Validating id set registration for {file_path}")
             if not self.id_set_validator.is_file_valid_in_set(file_path):
                 return False
+            if file_type == FileType.INCIDENT_TYPE:
+                if not self.id_set_validator.is_incident_type_using_real_playbook(file_path):
+                    return False
+            if file_type == FileType.SCRIPT:
+                if not self.id_set_validator.is_script_using_real_command(file_path):
+                    return False
```
Comment (roysagi) [inline demisto_sdk/commands/validate/validate_manager.py:349]: I think these would fit better in the `id_set_validator.is_file_valid_in_set` method - you can pass the file_type there to save up time.

**Review by roysagi — CHANGES_REQUESTED** [review]

File: demisto_sdk/commands/common/hook_validations/id.py (hunk 2 above)
Comment (bakatzir) [reply demisto_sdk/commands/common/hook_validations/id.py:70]: not required, but better to indicate that this func is private.

**Comment by bakatzir** [comment]
Not merging till we will know that the content is valid: https://github.com/demisto/content/pull/10766

**Comment by bakatzir** [comment]
id set validations on msater are green: https://app.circleci.com/pipelines/github/demisto/content/58518/workflows/7d5d9eae-d4d9-4425-9e7b-487620c399ab/jobs/244449 @yuvalbenshalom FYI.
merging.
